### PR TITLE
[OBSDEF-7281] - added datagrid filter popup position

### PR DESCRIPTION
**Pull request by vinodwani18**
Description:
Added new prop for the filter position.

Jira:
https://jira.cec.lab.emc.com/browse/OBSDEF-7281

UI screen shot:
image

### CODE REVIEW

File: src/datagrid/DataGridFilter.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ import {DebounceUtils} from "../forms/common/DebounceUtils";
  * @param {disabled} boolean value to enable or disable filter
  * @param {debounce} boolean value to apply debounce behaviour
  * @param {debounceTime} number value debounceTime/Delay value in miliseconds
+ * @param {position} number position of the filter popup
```
Comment (akate1) [inline src/datagrid/DataGridFilter.tsx:30]: number position ->   position

File: src/datagrid/DataGridFilter.tsx [hunk 2]
```diff
@@ -62,6 +64,12 @@ export enum FilterType {
     CUSTOM = "Custom",
 }
 
+// Enum for filter position
+export enum FilterPosition {
+    LEFT = "left",
+    MIDDLE = "middle",
```
Comment (akate1) [inline src/datagrid/DataGridFilter.tsx:70]: can we rename MIDDLE to CENTER ?? Just a thought

**Review by akate1 — APPROVED** [review]
LGTM

File: src/datagrid/DataGridFilter.tsx (hunk 2 above)
Comment (maheshkulkarni4) [reply src/datagrid/DataGridFilter.tsx:70]: Lets call it "Center"

File: src/datagrid/DataGridFilter.tsx [hunk 3]
```diff
@@ -228,6 +237,14 @@ export class DataGridFilter extends React.PureComponent<DataGridFilterProps, Dat
             }
             return child;
         });
+        let alignment;
+        if (position === FilterPosition.RIGHT) {
+            alignment = "195px";
```
Comment (maheshkulkarni4) [inline src/datagrid/DataGridFilter.tsx:242]: Can we use rems instead ?  Verify that these filters render properly when you resize your screens.  It should always stick to the column where Filter has opened. 

Check following:

1) open filter
2) resize your screen
3) check that filter stays with the column

File: src/datagrid/DataGridFilter.tsx [hunk 4]
```diff
@@ -228,6 +237,14 @@ export class DataGridFilter extends React.PureComponent<DataGridFilterProps, Dat
             }
             return child;
         });
+        let alignment;
+        if (position === FilterPosition.RIGHT) {
```
Comment (maheshkulkarni4) [inline src/datagrid/DataGridFilter.tsx:241]: use switch case instead ?  treat left as default case

**Review by maheshkulkarni4 — APPROVED** [review]
LGTM

File: src/datagrid/DataGridFilter.tsx (hunk 2 above)
Comment (vinodwani18) [reply src/datagrid/DataGridFilter.tsx]: done

File: src/datagrid/DataGridFilter.tsx (hunk 1 above)
Comment (vinodwani18) [reply src/datagrid/DataGridFilter.tsx]: done

File: src/datagrid/DataGridFilter.tsx (hunk 3 above)
Comment (vinodwani18) [reply src/datagrid/DataGridFilter.tsx]: done

File: src/datagrid/DataGridFilter.tsx (hunk 4 above)
Comment (vinodwani18) [reply src/datagrid/DataGridFilter.tsx]: done

**Pull request merged**